src/nvim/api/vim.c

```diff
@@ -812,17 +813,9 @@ void nvim_set_current_dir(String dir, Error *err)
   memcpy(string, dir.data, dir.size);
   string[dir.size] = NUL;
 
-  try_start();
-
-  if (vim_chdir((char_u *)string)) {
-    if (!try_end(err)) {
-      api_set_error(err, kErrorTypeException, "Failed to change directory");
-    }
-    return;
-  }
-
-  post_chdir(kCdScopeGlobal, true);
-  try_end(err);
+  char_u *cmd = concat_str((const char_u *)"cd ", (const char_u *)string);
+  nvim_exec(cstr_to_string(vim_strsave_fnameescape((const char *)cmd, false)), false, err);
```

Comment:
the `vim_strsave_fnameescape` result needs to be freed.

asan failure:


```
2021-09-26T23:57:07.6046138Z = Direct leak of 13 byte(s) in 1 object(s) allocated from:
2021-09-26T23:57:07.6046709Z =     #0 0x4d8e9d in malloc ??:?
2021-09-26T23:57:07.6047354Z =     #1 0x10ff222 in try_malloc /home/runner/work/neovim/neovim/src/nvim/memory.c:74:15
2021-09-26T23:57:07.6048162Z =     #2 0x10ff41c in xmalloc /home/runner/work/neovim/neovim/src/nvim/memory.c:108:15
2021-09-26T23:57:07.6049060Z =     #3 0x188b7ef in vim_strsave_escaped_ext /home/runner/work/neovim/neovim/src/nvim/strings.c:108:28
2021-09-26T23:57:07.6049996Z =     #4 0x188b42b in vim_strsave_escaped /home/runner/work/neovim/neovim/src/nvim/strings.c:79:10
2021-09-26T23:57:07.6050944Z =     #5 0xd2889f in vim_strsave_fnameescape /home/runner/work/neovim/neovim/src/nvim/ex_getln.c:4131:13
2021-09-26T23:57:07.6051864Z =     #6 0x6d2517 in nvim_set_current_dir /home/runner/work/neovim/neovim/src/nvim/api/vim.c:817:28
2021-09-26T23:57:07.6054155Z =     #7 0x5d54a2 in handle_nvim_set_current_dir /home/runner/work/neovim/neovim/build/src/nvim/auto/api/private/dispatch_wrappers.generated.h:4050:3

Reply:
`cstr_to_string` also calls `xmemdupz` so there are two strings that need to be freed. Probably should just use `cstr_as_string` instead.